libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -630,6 +630,104 @@ string YulUtilFunctions::overflowCheckedIntExpFunction(
 	});
 }
 
+string YulUtilFunctions::overflowCheckedIntLiteralExpFunction(
+	RationalNumberType const& _baseType,
```

Comment:
Passing the RationalNumberType type looks overly complex to me. Can you modify RationalNumberType to provide public access to its `m_value` member and pass that as a rational / bigint instead to this function?

Reply:
Almost the same reason as https://github.com/ethereum/solidity/pull/9816#discussion_r495004290. We want to know if the type is negative or not and also the value in two's complement. I think skipping the type will lead to more code.